### DISCUSSION TITLE
How to react on a message published by MQTT broker in BSB-LAN code?

### DISCUSSION THREAD

**Discussion by lemon75** [fredlcore/BSB-LAN#591] [Q&A] [answered]
Hello all,
tried to find anything in the existing discussion but did not succeed.

**I wonder what would be the smartest way to react on a message that my MQTT broker publishes.
I would like to set / reset a GPIO pin depending on this message.**

1) use BSB_LAN_custom.h
2) modify the MQTT part in BSB_LAN.ino

btw: it will be the same MQTT broker which is used for logging.

I'd be glad for any suggestions on this topic.
Thanks in advance,
Jan

**Comment by fredlcore** [comment, marked as answer]
Isn't it possible to do this "reaction" within the home automation software and send an URL command instead (/G32=1 for example)? If not, then I definitively would prefer to go the route via BSB_LAN_custom.h because these changes would survive an update of the main code.
I wouldn't be against changes in the code to act on MQTT messages with regard to the GPIOs in the same way as BSB-LAN reacts to MQTT messages for parameters, so if you have to go down this way, you could provide us with a PR and I'd have a look at whether there is a chance to add this to the main code.

**Reply by lemon75** [reply]
@fredlcore, thank you for the quick answer.
You are right, the HTML-Codes already offer the possibility that I need. Somehow I must have overlooked this option 🙄
And yes, the Smart-Home system will issue this command via mqtt broker to BSB-LAN.

But maybe I do not need this anymore at all. In the other discussion we both replied to yesterday (heating the buffer) you mentioned the "trick" using the action type ("Wirksinn") which might do the job for me too.

In my case the elco heat pump can load the whole buffer (800 L) and not only the upper part of it in case of surplus PV energy. Together with a higher set-value of the temperature this offers quite a lot of buffering 🙂

Thanks,
Jan

**Comment by fredlcore** [comment]
Ok, glad to hear if that solution would work for you two. If it does, please let us know how you did it in the other discussion thread so that others can benefit from it, too!
